swap/tests/testutils/mod.rs

```diff
@@ -391,21 +404,163 @@ where
     testfn(test).await
 }
 
+fn random_prefix() -> String {
+    use rand::Rng;
+    const CHARSET: &[u8] = b"abcdefghijklmnopqrstuvwxyz";
+    const LEN: usize = 8;
+    let mut rng = rand::thread_rng();
+
+    let prefix: String = (0..LEN)
+        .map(|_| {
+            let idx = rng.gen_range(0, CHARSET.len());
+            CHARSET[idx] as char
+        })
+        .collect();
+    prefix
+}
+
 async fn init_containers(cli: &Cli) -> (Monero, Containers<'_>) {
-    let bitcoind = Bitcoind::new(&cli, "0.19.1").unwrap();
-    let _ = bitcoind.init(5).await;
+    let bitcoind_name = format!("{}{}", random_prefix(), "bitcoind");
+    let volume = random_prefix();
+    let (bitcoind, bitcoind_url) =
+        init_bitcoind_container(&cli, volume.clone(), bitcoind_name.clone(), volume.clone())
+            .await
+            .expect("could not init bitcoind");
+    let electrs = init_electrs_container(&cli, volume.clone(), bitcoind_name, volume).await;
+    let (monero, monerods) = init_monero_container(&cli).await;
+    (monero, Containers {
+        bitcoind_url,
+        bitcoind,
+        monerods,
+        electrs,
+    })
+}
+
+async fn init_bitcoind_container(
+    cli: &Cli,
+    volume: String,
+    name: String,
+    network: String,
+) -> Result<(Container<'_, Cli, bitcoind::Bitcoind>, Url)> {
+    let image = bitcoind::Bitcoind::default()
+        .with_mapped_port(Port {
+            local: 7041,
+            internal: 18443,
+        })
```

Comment:
Afaik our bitcoind image exports the default ports and you could rely on docker to provide you random ports. That way you would not run into random `address in use` issues. 
See https://github.com/testcontainers/testcontainers-rs/blob/fe51e06fb2c44b1d3e3fd627cedea3fd582487ef/src/core/container.rs#L81